cli/azd/cmd/auth_token.go

```diff
@@ -76,8 +86,36 @@ func (a *authTokenAction) Run(ctx context.Context) (*actions.ActionResult, error
 
 	var cred azcore.TokenCredential
 
+	// 1) flag --tenant-id is the highest priority. If it is not use, azd will check if subscriptionId is set as env var
+	tenantId := a.flags.tenantID
+	// 2) try to resolve tenant id when AZURE_SUBSCRIPTION_ID is set as system env
+	if tenantId == "" {
+		if subIdAtSystemEnv, found := os.LookupEnv(environment.SubscriptionIdEnvVarName); found {
+			if resolvedTenantId, err := a.subResolver.LookupTenant(ctx, subIdAtSystemEnv); err == nil {
+				tenantId = resolvedTenantId
+			} else {
+				log.Println("Found AZURE_SUBSCRIPTION_ID in system env, but azd couldn't find the Azure directory where" +
+					" it belongs. The AZURE_SUBSCRIPTION_ID is ignored.")
+			}
+		}
+	}
+	// 3) last try to resolve tenantId. This time, check if AZURE_SUBSCRIPTION_ID is set within the current azd env
```

Comment:
This doesn't quite match my expectation of what this should be doing. Repasting from different thread to be clear:

`auth_token` should be doing:

LoadTenantFromEnv:
1. Load context. If not found, move along (not azd project).
1. Load default environment name. If not found, move along (no env was configured to be default, must be set explicitly).
2. If default environment name is set, load environment. Error if environment isn't found (invalid setup).
3. If environment is found, load subscription. Error out if subscription is not set (invalid setup).
4. If subscription it set, resolve tenant. Error out if resolve fails (invalid setup).

Reply:
It's only 3 alternatives.  It is described here: https://github.com/Azure/azure-dev/issues/1712#issuecomment-1464654397